.pre-commit-config.yaml

```diff
@@ -7,7 +7,7 @@ repos:
     -   id: trailing-whitespace
 -   repo: https://github.com/astral-sh/ruff-pre-commit
     # Ruff version.
-    rev: v0.2.1
+    rev: v0.5.1
```

Comment:
Should we bump the min version in `_min_dependencies` and update the lock files then ?

Otherwise I can imagine that a user has an older version and then the CI complains for errors that he doesn't see locally.

Reply:
This PR also updates `_min_dependencies`, should I try to recreate the lock files? I don't mind, but that happens anyway the monday after we merge?

Reply:
Oh I missed that. For the lock files, it should be fine because it's just the linter, but usually it's better to regenerate in the PR updating the dependencies so that we can see and fix any issue right away.